docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,433 @@
+        value = newValue;
+        emit ValueChanged(newValue);
+    }
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+}
+
+```
+
+
+## Transparent upgradable proxies
+
+Transparent upgradable proxies provide a way to upgrade a smart contract without changing its address or requiring any change in the contract's interaction code. With transparent proxies, a contract's address is owned by a proxy contract, which forwards all calls to the actual contract implementation. When a new implementation is deployed, the proxy can be upgraded to point to the new implementation, allowing for seamless upgrades without requiring changes to the contract's interaction code.
+
+To deploy a simple upgradable contract on the zkSync local setup, first create a test wallet and add it to the new Deployer.
+
+```typescript
+
+  const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+  const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+  const deployer = new Deployer(hre, zkWallet);
+```
+
+After that, simply load the Box artifact and call the `deployProxy` method from the `zkUpgrades` hre property.
+
+```typescript
+  const contract = await deployer.loadArtifact(contractName);
+  await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+```
+
+And that is it! <br/>
+The `deployProxy` method will deploy your implementation contract on the zkSync network, deploy the proxy admin contract, and finally, deploy the transparent proxy.
+
+Here is one example of the whole deploy proxy script:
+
+```typescript
+
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { Wallet } from 'zksync-web3';
+
+const hre = require('hardhat');
+
+async function main() {
+    const contractName = 'Box';
+    console.log('Deploying ' + contractName + '...');
+
+    const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+    const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+    const deployer = new Deployer(hre, zkWallet);
+
+    const contract = await deployer.loadArtifact(contractName);
+    const box = await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+
+    await box.deployed();
+    console.log(contractName + ' deployed to:', box.address);
+
+    box.connect(zkWallet);
+    const value = await box.retrieve();
+    console.log('Box value is: ', value.toNumber());
+}
+
+main();
+
+
+```
+
+> **_IMPORTANT NOTE:_** `deployProxy` method (and other deploy/upgrade methods from the zkUpgrades) needs the know which wallet to use to deploy smart contracts. For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network. This provider is configured in the hardhat config file, by stating the RPC url of the network to connect to.
```

Comment:
```suggestion
:::warning Important 
- The `deployProxy` method (and other deploy/upgrade methods from the zkUpgrades) needs to know which wallet to use to deploy smart contracts. 
- For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network.
- The provider is configured in the hardhat config file with the RPC URL of the network.
```

Reply:
Would it be better to provide the full hardhat config file instead of just a snippet of it?

Reply:
Are you sure we want to remove the 'note' label from this text and move it under the 'hardhat config' subtitle? I left it in the previous section on purpose because it was directly related to the code and the way '.deployProxy' method receives 'deployer.zkWallet' instead of just 'zkWallet'.

Regarding the full config, I would say there is no need to specify it here, because it is the same hardhat config as stated at the top of this page. I changed the wording as a suggestion on how it can be stated more explicitly. Let me know if that is sufficient.

Reply:
No, I did not want to do that. Keep the note, but make it a warning instead ... 

:::warning
- `deployProxy` method (and other deploy/upgrade methods from the zkUpgrades) needs to know which wallet to use to deploy smart contracts. 
- For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network. 
- This provider is configured in the hardhat config file, by stating the RPC url of the network to connect to.
:::

Reply:
And SGTM.